core/src/main/java/org/eclipse/hono/util/RegistrationConstants.java

```diff
@@ -158,17 +160,20 @@ public static JsonObject getRegistrationJson(final String action, final String t
     }
 
     public static JsonObject getRegistrationJson(final String action, final String tenantId, final String deviceId, final JsonObject payload) {
-        return getRegistrationJson(action, tenantId, deviceId, null, payload);
+        return getRegistrationJson(action, tenantId, deviceId, null, null, payload);
     }
 
-    public static JsonObject getRegistrationJson(final String action, final String tenantId, final String deviceId, final String key, final JsonObject payload) {
+    public static JsonObject getRegistrationJson(final String action, final String tenantId, final String deviceId, final String key, final String value, final JsonObject payload) {
         final JsonObject msg = new JsonObject();
         msg.put(APP_PROPERTY_ACTION, action);
         msg.put(APP_PROPERTY_DEVICE_ID, deviceId);
         msg.put(APP_PROPERTY_TENANT_ID, tenantId);
         if (key != null) {
             msg.put(APP_PROPERTY_KEY, key);
         }
+        if (value != null) {
```

Comment:
At the moment you can register complex values. I do not know what you had in mind for the device registry. If there should be complex values, it might okay to look at them more as 'document' properties, which are not searchable. If we want to search vor this (in future) I think we need a query language and I would not suggest to define this in the API if it is not really necessary.